src/parser/deathknight/frost/modules/features/BreathOfSindragosa.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+
+import Analyzer from 'parser/core/Analyzer';
+
+class BreathOfSindragosa extends Analyzer{
+  static dependancies = {
+
+  }
+
+  beginTimestamp = 0;
+  casts = 0;
+  badCasts = 0;
+  totalDuration = 0;
+  breathActive = false;
+
+  on_byPlayer_applybuff(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.BREATH_OF_SINDRAGOSA_TALENT.id){
+      return;
+    }
+    this.casts += 1;
+    this.beginTimestamp = event.timestamp;
+    this.breathActive = true;
+  }
+
+  on_byPlayer_removebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.BREATH_OF_SINDRAGOSA_TALENT.id){
+      return;
+    }
+    this.breathActive = false;
+    const duration = event.timestamp - this.beginTimestamp;
+    if(duration < 15000){
+      this.badCasts +=1;
+    }
+    this.totalDuration += duration;
+  }
+
+  on_finished(event){
+    if(this.breathActive){
+      this.endTimestamp = event.timestamp;
+      const duration = event.timestamp - this.beginTimestamp;
+      if(duration < 20000){
```

Comment:
Shouldn't this be the same as above? (15000 ms).

Reply:
Actually it probably shouldn't even there.  A cast still going on fight end shouldn't even be counted in this module since you can't trust the duration